packages/data-service/src/data-service.ts

```diff
@@ -1425,60 +1421,48 @@ export class DataServiceImpl implements DataService {
     return this._collection(ns, 'CRUD').find(filter, options);
   }
 
-  findOneAndReplace(
+  async findOneAndReplace(
     ns: string,
     filter: Filter<Document>,
     replacement: Document,
-    options: FindOneAndReplaceOptions,
-    callback: Callback<Document>
-  ): void {
+    options: FindOneAndReplaceOptions
+  ): Promise<Document | null> {
     const logop = this._startLogOp(
       mongoLogId(1_001_000_044),
       'Running findOneAndReplace',
       { ns }
     );
     const coll = this._collection(ns, 'CRUD');
-    callbackify(allArgumentsMandatory(coll.findOneAndReplace.bind(coll)))(
-      filter,
-      replacement,
-      options,
-      (error, result) => {
-        logop(error);
-        if (error) {
-          // @ts-expect-error Callback without result...
-          return callback(this._translateErrorMessage(error));
-        }
-        callback(null, result.value!);
```

Comment:
We had tests in CRUD that were expecting this to return null, so I wasn't sure why we were asserting here and providing a type that doesn't expect null value returned from the driver, changed this to align with that CRUD tests were expecting